File: Server/src/main/scala/jk_5/nailed/util/NailedFoodStats.java
package jk_5.nailed.util;

import net.minecraft.entity.player.*;
import net.minecraft.util.*;

/**
 * Created by matthias on 5/8/14.
 */
public class NailedFoodStats extends FoodStats {

    private int minFoodLevel = 0;
    private int maxFoodLevel = 20;

    @Override
    public void onUpdate(EntityPlayer player) {
        super.onUpdate(player);
        if(this.maxFoodLevel < this.minFoodLevel && this.maxFoodLevel > 0){
            return;
        }
        if(this.getFoodLevel() < this.minFoodLevel){
            this.foodLevel = this.minFoodLevel;
        }
        if(this.getFoodLevel() > this.maxFoodLevel && this.maxFoodLevel > 0){
            this.foodLevel = this.maxFoodLevel;
        }
    }

    public void setMinFoodLevel(int value) {
        this.minFoodLevel = value;
    }

    public int getMinFoodLevel() {
        return this.minFoodLevel;
    }

    public void setMaxFoodLevel(int value) {
        this.maxFoodLevel = value;
    }

    public int getMaxFoodLevel() {
        return this.maxFoodLevel;
    }

    @Override
    public void setFoodLevel(int level) {
        this.foodLevel = level;
    }
}

File: Server/src/main/scala/jk_5/nailed/util/NailedFoodStats.scala
package jk_5.nailed.util

import net.minecraft.util.FoodStats
import net.minecraft.entity.player.EntityPlayer

/**
 * No description given
 *
 * @author jk-5
 */
class NailedFoodStats extends FoodStats {

  private var minFoodLevel = 0
  private var maxFoodLevel = 20

  override def onUpdate(player: EntityPlayer){
    super.onUpdate(player)
    if(this.maxFoodLevel < this.minFoodLevel && this.maxFoodLevel > 0) return
    if(this.foodLevel < this.minFoodLevel) this.foodLevel = this.minFoodLevel
    if(this.foodLevel > this.maxFoodLevel && this.maxFoodLevel > 0) this.foodLevel = this.maxFoodLevel
  }

  def setMinFoodLevel(value: Int) = this.minFoodLevel = value
  def getMinFoodLevel = this.minFoodLevel
  def setMaxFoodLevel(value: Int) = this.maxFoodLevel = value
  def getMaxFoodLevel = this.maxFoodLevel
  def setFood(level: Int) = this.foodLevel = level
}
